apps/prairielearn/src/pages/authLogin/authLogin.ts

```diff
@@ -69,4 +72,37 @@ router.get(
   }),
 );
 
+const DevLoginParamsSchema = z.object({
+  uid: z.string().nonempty(),
+  name: z.string().nonempty(),
+  uin: z.string().nullable().optional().default(null),
+});
+
+router.post(
+  '/',
+  asyncHandler(async (req, res, _next) => {
+    if (!config.devMode) {
+      throw error.make(404, 'Not Found');
+    }
+
+    if (req.body.__action === 'dev_login') {
+      const body = DevLoginParamsSchema.parse(req.body);
```

Comment:
I ❤️ it!!!

Reply:
I want to do much more of this 🙂

Reply:
Yes please! We should be doing this on every POST.